internal/plugin/github.go

```diff
@@ -4,56 +4,56 @@ import (
 	"io"
 	"net/http"
 	"net/url"
-	"strings"
 
-	"github.com/pkg/errors"
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cachehash"
 )
 
 type githubPlugin struct {
-	raw      string
-	org      string
-	repo     string
-	revision string
-	dir      string
+	RefLike
```

Comment:
instead of embedding the struct, could we make it an explicit field to distinguish methods/fields that are from RefLike versus native to `githubPlugin`?

But its tolerable if it means having to re-implement a lot of methods unnecessarily

Reply:
I can make that change. It does mean that `plugin.Repo` becomes `plugin.ref.Repo` so the code is a big uglier. (There's roughly 12 callsites in this file that would change)